### DISCUSSION TITLE
getTransactionError: Cannot read properties of null (reading 'gasLimit')

### DISCUSSION THREAD

**Discussion by HenryDotApp** [wevm/viem#1455] [Question]
Running into this error while using Viem with Polygon, but it seems to happen very flakily. A similar error was reported months ago but it never got any follow-up from the original asker, so I'm asking this here.

For context of my usage, `sendTransaction` is getting called here: https://github.com/reservoirprotocol/reservoir-kit/blob/main/packages/sdk/src/utils/viemWallet.ts#L69

The error log includes this bit:
```
Details: Cannot read properties of null (reading 'gasLimit')
Version: viem@1.10.8
    at getTransactionError (getTransactionError.js:8:1)
    at sendTransaction (sendTransaction.js:121:34)
    at async Object.handleSendTransactionStep (index.mjs:157:1)
    at async $a2481957a3362541$export$2d85f13e6286dded (index.mjs:327:1)
    at async eval (index.mjs:637:1)
```

I see in Viem's repo that `.gasLimit` is called here: https://github.com/wagmi-dev/viem/blob/339fbf1323fb38c1c5b4ab7388577327f8aee682/src/utils/formatters/block.ts#L54 but it doesn't make sense why that line would throw, and not an earlier line, if `block` is null

Please let me know what other info I can provide that would help narrow this down! One thing that would greatly help is understanding where else `.gasLimit` could be being called

**Comment by pedromcunha** [comment]
@jxom got any idea what's happening here? Your insight could be really helpful, thanks in advance 🙏

**Comment by HenryDotApp** [comment]
@jxom Bump :) Still seeing this issue intermittently

**Comment by karacurt** [comment]
I am also having a similar issue like this! I really would love to know if this was already solved!

**Reply by BravoNatalie** [reply]
try to use the `gas: bigint` parameter for the gas limit. 
https://viem.sh/docs/accounts/signTransaction#gas-optional
